app/vmagent/remotewrite/client.go

```diff
@@ -510,6 +538,24 @@ func getRetryDuration(retryAfterDuration, retryDuration, maxRetryDuration time.D
 	return retryDuration
 }
 
+var (
+	repackBufPool bytesutil.ByteBufferPool
+)
+
+func repackBlockFromZstdToSnappy(block []byte) ([]byte, error) {
```

Comment:
I'd rename this function to `mustRepackBlockFromZstdToSnappy(block []byte) []byte`, and panic inside this function when it cannot decompress the `block`, since this error means either programming error with improper usage of this function (aka BUG) or corruption of the `block` contents. Both cases need immediate attention from the operator / software engineer, so the best way to do this is to panic with the information, which can help quickly resolving the issue.